app/assets/javascripts/dashboard.js

```diff
@@ -1,33 +1,83 @@
 MinionPoller = {
-    poll: function() {
-        this.request();
-        return setInterval(this.request, 5000);
-    },
-
-    request: function() {
-        $.ajax({
-            url: $('#nodes').data('url'),
-            dataType: "json",
-            success: function(data) {
-                var rendered = "";
-
-                for (i = 0; i < data.length; i++) {
-                    rendered += MinionPoller.render(data[i]);
-                }
-                $("#nodes").html(rendered);
-            }
-        });
-    },
-
-    render: function(minion) {
-        return "<div> \
-  <a href=\"/nodes/" + minion.id + "\">" + minion.hostname + "</a>\
-  <dl>\
-    <dt>Created at</dt>\
-    <dd>" + minion.created_at + "</dd>\
-\
-    <dt>Updated at</dt>\
-    <dd>" + minion.updated_at + "</dd>\
-  </dl>"
+  poll: function() {
+    this.request();
+    return setInterval(this.request, 5000);
+  },
+
+  request: function() {
+    $.ajax({
+      url: $('.nodes-container').data('url'),
+      dataType: "json",
+      success: function(data) {
+        var rendered = "";
+        MinionPoller.selectedMaster = $("input[name='roles[master]']:checked").val();
+
+        minions = data.assigned_minions || data;
+
+        for (i = 0; i < minions.length; i++) {
+          if (MinionPoller.renderMode == "discovery") {
+            rendered += MinionPoller.renderDiscovery(minions[i]);
+          } else {
+            rendered += MinionPoller.render(minions[i]);
+          }
+        }
+        $(".nodes-container tbody").html(rendered);
```

Comment:
You can avoid rewriting the content of the node with so little checks: For example
https://github.com/kubic-project/velum/blob/97fdf6e13f0f27865f03a307edf94e7138c935b9/app/assets/javascripts/velumMinionPoller.js#L84